scala-generator/src/main/scala/models/KafkaConsumer.scala

```diff
@@ -18,6 +19,15 @@ object KafkaConsumer extends CodeGenerator {
     Right(generateCode(form))
   }
 
+  def generateTopicRegex(topicFn: String, apiVersion: String) = {
+    val apiVersionVariable = Seq("${apiVersion}", "$apiVersion").map(Regex.quote(_)).mkString("|")
+    val tenantVariable = Seq("${tenant}", "$tenant").map(Regex.quote(_)).mkString("|")
+    topicFn
+      .replaceAll(apiVersionVariable, Regex.quoteReplacement(apiVersion))
+      //`tenantsPattern` is a val defined in the `topicRegex` function, see `source` below.
```

Comment:
so when you call a function you get access to the local vars it has or am I missing something?